crates/sui-replay/src/config.rs

```diff
@@ -148,6 +144,34 @@ pub fn url_from_str(s: &str) -> Result<Uri, ReplayEngineError> {
     })
 }
 
+/// If rpc_url is provided, use it. Otherwise, load the network config from the config file.
+pub fn get_rpc_url(
+    rpc_url: Option<String>,
+    config_path: Option<PathBuf>,
+    chain: Option<String>,
+) -> String {
+    match rpc_url {
+        Some(url) => url,
+        None => {
+            let config_path = config_path
+                .map(|p| p.to_str().unwrap().to_string())
+                .unwrap_or_else(|| DEFAULT_CONFIG_PATH.to_string());
+            let chain = chain.unwrap_or("mainnet".to_string());
+            info!(
+                "RPC URL not provided. Loading network config for {:?} from config file {:?}. \
+                    If a different chain is desired, please provide the chain name.",
+                chain, config_path
+            );
+            ReplayableNetworkConfigSet::load_config(config_path)
+                .unwrap()
+                .get_base_config(&chain)
+                .unwrap_or_else(|| panic!("Unable to find network config for {:?}", chain))
```

Comment:
I'd simply return an error instead of panicking, and instruct the user to run sui client to generate a new config file.